### PR TITLE
Creating product - templateId wrong UUID - no validation

### PR DESCRIPTION
**Pull request by wiewiurdp**

### CODE REVIEW

File: module/product/src/Infrastructure/Validator/TemplateExists.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * @Annotation
+ */
+class TemplateExists extends Constraint
+{
+    /**
+     * @var string
+     */
+    public $message = 'Template {{ value }} doesn\'t exists.';
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Validator/TemplateExists.php]: Without dot at end

File: module/product/src/Infrastructure/Validator/TemplateExistsValidator.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+     * @param mixed      $value
+     * @param Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof TemplateExists) {
+            throw new UnexpectedTypeException($constraint, TemplateExists::class);
+        }
+
+        if (null === $value || '' === $value) {
```
Comment (BastekBielawski) [inline module/product/src/Infrastructure/Validator/TemplateExistsValidator.php]: Now you allow to set empty template ID

File: module/product/src/Infrastructure/Validator/TemplateExists.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types = 1);
+
+namespace Ergonode\Product\Infrastructure\Validator;
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ */
+class TemplateExists extends Constraint
```
Comment (rprzedzik) [inline module/product/src/Infrastructure/Validator/TemplateExists.php]: This constraint should be in designer module

File: module/product/src/Application/Model/ProductCreateFormModel.php [hunk 4]
```diff
@@ -36,6 +37,7 @@ class ProductCreateFormModel
      *
      * @Assert\NotBlank(message="Template is required")
      * @Assert\Uuid()
```
Comment (BastekBielawski) [inline module/product/src/Application/Model/ProductCreateFormModel.php]: Should be ``Uuid(versions=[5])``

File: module/designer/src/Infrastructure/Validator/TemplateExistsValidator.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            return;
+        }
+
+        if (!is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedTypeException($value, 'string');
+        }
+
+        $value = (string) $value;
+        try {
+            $template = $this->templateRepository->load(new TemplateId($value));
```
Comment (BastekBielawski) [inline module/designer/src/Infrastructure/Validator/TemplateExistsValidator.php]: We should change it.
``
$template = false;
if (TemplateId::isValid($value)) {
    $template = $this->templateRepository->load(new TemplateId($value));
}
``

**Pull request merged**